api/task_processor/processor.py

```diff
@@ -25,6 +25,10 @@ def run_tasks(num_tasks: int = 1) -> typing.List[TaskRun]:
         task_runs = []
 
         for task in tasks:
+            if task.run_every is not None and not task.is_next_run_scheduled:
```

Comment:
The other slight concern is that the code at the top of the `_run_task()` function is double checking that the task hasn't already been run. Here, if that is the case, we'd end up with 2 scheduled tasks for the future, right?

Reply:
> The other slight concern is that the code at the top of the `_run_task()` function is double checking that the task hasn't already been run. Here, if that is the case, we'd end up with 2 scheduled tasks for the future, right?

Are you talking about the current solution?

Reply:
Yes, because you're scheduling the next task before calling `_run_task` so the check to verify we haven't already run the task (and subsequently scheduled a future task) isn't happening until you've already scheduled the future task. 

... ah, but I guess that the check at `Task.is_next_run_scheduled` should cover us for that case?

Reply:
> ... ah, but I guess that the check at Task.is_next_run_scheduled should cover us for that case?

Yep, exactly